jaeger_client/span.py

```diff
@@ -142,6 +142,22 @@ def set_baggage_item(self, key, value):
     def get_baggage_item(self, key):
         return self.context.baggage.get(key)
 
+    def remove_baggage_item(self, key):
```

Comment:
the API of the Span class is dictated by the OpenTracing API, we cannot arbitrarily add new methods. I believe if we simply implement the change to the SpanContext below, then you can achieve the same effect of removing baggage item via `span.set_baggage_item(key, None)`.